msg_helpers.go

```diff
@@ -612,6 +613,76 @@ func packDataNsec(bitmap []uint16, msg []byte, off int) (int, error) {
 	return off, nil
 }
 
+func unpackDataSVCB(msg []byte, off int) ([]SVCBKeyValue, int, error) {
+	var xs []SVCBKeyValue
+	var code uint16
+	var length uint16
+	var err error
+	for off < len(msg) {
+		code, off, err = unpackUint16(msg, off)
+		if err != nil {
+			return nil, len(msg), &Error{err: "overflow unpacking svcb"}
+		}
+		length, off, err = unpackUint16(msg, off)
+		if err != nil || off+int(length) > len(msg) {
+			return nil, len(msg), &Error{err: "overflow unpacking svcb"}
+		}
+		e := makeSVCBKeyValue(code)
+		if e == nil {
+			return nil, len(msg), &Error{err: "svcb invalid key"}
+		}
+		if err := e.unpack(msg[off : off+int(length)]); err != nil {
+			return nil, len(msg), err
+		}
+		xs = append(xs, e)
+		off += int(length)
+	}
+
+	prev := uint16(0)
+	for _, e := range xs {
+		if e.Key() <= prev {
+			return nil, len(msg), &Error{err: "svcb keys not in strictly increasing order"}
+		}
+		prev = e.Key()
+	}
+	return xs, off, nil
+}
+
+func packDataSVCB(originalPairs []SVCBKeyValue, msg []byte, off int) (int, error) {
+	pairs := make([]SVCBKeyValue, len(originalPairs))
+	copy(pairs, originalPairs)
+	sort.Slice(pairs, func(i, j int) bool {
+		return pairs[i].Key() < pairs[j].Key()
+	})
+	prev := uint16(0)
+	for _, e := range pairs {
+		if e.Key() == prev {
+			return len(msg), &Error{err: "repeated svcb keys are not allowed"}
+		}
+		prev = e.Key()
+	}
+	for _, el := range pairs {
+		packed, err := el.pack()
+		if err != nil {
+			return len(msg), err
+		}
+		if len(packed) > 65535 {
```

Comment:
why this check?

Reply:
Two-byte limit to encode value length? Right, removing.